packages/imperative/src/cnfg/cmd/set/set.handler.ts

```diff
@@ -0,0 +1,71 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { ICommandHandler, IHandlerParameters } from "../../../../../cmd";
+import { Config, IConfigOpts } from "../../../../../config";
+import { ImperativeError } from "../../../../../error";
+import { CredentialManagerFactory } from "../../../../../security";
+import { ImperativeConfig } from "../../../../../utilities";
+
+/**
+ * The get command group handler for cli configuration settings.
+ *
+ */
+export default class SetHandler implements ICommandHandler {
+
+    /**
+     * Process the command and input.
+     *
+     * @param {IHandlerParameters} params Parameters supplied by yargs
+     *
+     * @throws {ImperativeError}
+     */
+    public async process(params: IHandlerParameters): Promise<void> {
+
+        // Setup the credential vault API for the config
+        let opts: IConfigOpts = null;
+        if (CredentialManagerFactory.initialized) {
+            opts = {
+                vault: {
+                    load: ((k: string): Promise<string> => {
+                        return CredentialManagerFactory.manager.load(k)
+                    }),
+                    save: ((k: string, v: any): Promise<void> => {
+                        return CredentialManagerFactory.manager.save(k, v);
+                    }),
+                    name: CredentialManagerFactory.manager.name
+                }
+            };
+        }
+
+        // Create the config, load the secure values, and activate the desired layer
+        const config = Config.load(ImperativeConfig.instance.rootCommandName, opts);
+        await config.api.secure.load();
+        config.api.layers.activate(params.arguments.user, params.arguments.global);
+
+        // Get the value to set
+        let value = params.arguments.value;
+        if (params.arguments.json) {
+            try {
+                value = JSON.parse(value);
+            } catch (e) {
+                throw new ImperativeError({ msg: `could not parse JSON value: ${e.message}` });
+            }
+        }
+
+        // Set the value in the config, save the secure values, write the config layer
+        config.set(params.arguments.property, params.arguments.value, {
+            secure: params.arguments.secure,
```

Comment:
This PR looks awesome!

With the current implementation, is it true that if a user forgets to provide the `--secure` flag, they could accidentally store a credential in plain text? Currently, once the SCS plugin is installed, any fields designated as secure in profile schemas will always be stored securely.

Reply:
Thanks @tjohnsonBCM!

That is correct. `config` profiles are simple key/value pairs and there is no concept of "profile schema". 

The intention is to create the config by hand. However, I wanted a generic mechanism to set secure fields... So funnily enough, setting secure fields is the reason `zowe cnfg set` exists.

I don't see why someone would prefer `zowe cnfg set` over editing by hand (or editing the JSON programmatically), so perhaps we could replace `zowe cnfg set` with something like `zowe cnfg secure <property path> <value>`? 

I think that would eliminate potential confusion... i.e. why would one even use `zowe cnfg set`? ... and hopefully addresses your concern of "accidentally" saving sensitive info in plain text.

What do you think? 

Also, @dkelosky was looking for a mechanism to completely create the config by hand and then secure fields specified in `secure`. I'd still like to support the request, but after a config is created/secured, we'll need a mechanism to update secure fields (for example, password expiration).

Reply:
I like the idea of naming the command `zowe cnfg secure` to eliminate potential confusion.

Perhaps instead of removing `zowe cnfg set`, it could be kept for setting non-secure settings. While I agree that it would normally be preferrable to edit JSON manually or programmatically, I could see value in having commands similar to NPM for shell scripts to use: `zowe cnfg get`, `zowe cnfg set`, `zowe cnfg delete`, etc.